### PR TITLE
Document the current release process for Chef Server

### PR DESCRIPTION
**Pull request by stevendanna**

### CODE REVIEW

**Comment by stevendanna** [comment]
cc @chef/lob @charlesjohnson @schisamo 

I've taken a stab at documenting our *current* release process in as much detail as I understood it.  I believe that @tylercloke nearly has automated update tests ready to be integrated into this process.  I was unsure how the rehearsal cluster fits in.

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+- [ ] Once committed to master, an automated process will make the
+  change to the live website; however, it might take a few minutes for
+  the deploy to complete and then a few more for CDN caches to expire.
+
+- [ ] Publish your draft blog post.
+
+- [ ] Send an email to the chef and chef-dev mailing lists announcing
+the release with a link to the blog post.
+
+Chef Server is now released.
```
Comment (joedevivo) [inline RELEASE_PROCESS.md]: ### Post-Release

Increment the patch release number in all `chef-server` version files (e.g. 12.1.2 becomes 12.1.3)

* omnibus/config/projects/chef-server.rb
* src/chef-mover/relx.config
* src/chef-mover/src/mover.app.src
* src/oc_bifrost/apps/bifrost/src/bifrost.app.src
* src/oc_bifrost/rel/reltool.config (probably not, will confirm)
* src/oc_bifrost/relx.config
* src/oc_erchef/rel/reltool.config (probably not, will confirm)
* src/oc_erchef/relx.config
* src/oc_erchef/src/oc_erchef.app.src

**Comment by stevendanna** [comment]
@tylercloke Mind updating this with whatever adjustments you had to make on the last deploy (or leaving a comment about them and I can write them up and then merge this)

**Comment by tylercloke** [comment]
@stevendanna I mostly followed the doc verbatim except my upgrade testing via Delivery. I'm not sure if we need that pipeline to be "stable" before we consider it part of the release process. I am open to either putting it in now, or waiting another few weeks for it to get super stable (almost there).

File: RELEASE_PROCESS.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+release process and can be used by any team member to perform a
+release.
+
+## Pre-requisites
+
+In order to release, you will need the following accounts/permissions:
+
+- Local checkouts of the chef-server and chef-web-downloads repositories
+- Push access to the chef-server github repository
+- Chef Software, Inc Hipchat account
```
Comment (schisamo) [inline RELEASE_PROCESS.md]: We've integrated Lita into Slack so promotions can now be performed in the #engineering-services channel in Slack.

**Comment by tylercloke** [comment]
Oh other thing, let's add to the blog post section that puts a link to the downloads page and maybe the update docs.

**Comment by stevendanna** [comment]
I've gone ahead and merged this with a few changes.  @chef/lob please feel free to update as appropriate.

**Pull request merged**